### ISSUE TITLE
About interleaved training data

### ISSUE THREAD

**Issue by hychaochao** [ByteDance-Seed/Bagel#72]
Congratulations on Bagel — very nice work!

We noticed that you’ve released an example training dataset in [bagel_example.zip](https://lf3-static.bytednsdoc.com/obj/eden-cn/nuhojubrps/bagel_example.zip), which has been very helpful. However, we couldn’t find an example of the vision-text interleaved data format that you mentioned in the paper as part of Bagel’s training data.

Can you also share an example of this interleaved data format, along with some guidance on how to train Bagel using such data?  Thank u!!

**Comment by Andy1621** [comment]
@hychaochao  Thank you for your question! The editing dataset is actually an example of an interleaved dataset, since we provide multi-round data. For a general dataset, the logic is as follows:

https://github.com/ByteDance-Seed/Bagel/blob/cec368eedead3afc9f6c9c7bce3da0b86ce93ec9/data/interleave_datasets/edit_dataset.py#L53-L71

**Comment by hychaochao** [comment]
> [@hychaochao](https://github.com/hychaochao) Thank you for your question! The editing dataset is actually an example of an interleaved dataset, since we provide multi-round data. For a general dataset, the logic is as follows:
> 
> [Bagel/data/interleave_datasets/edit_dataset.py](https://github.com/ByteDance-Seed/Bagel/blob/cec368eedead3afc9f6c9c7bce3da0b86ce93ec9/data/interleave_datasets/edit_dataset.py#L53-L71)
> 
> Lines 53 to 71 in [cec368e](/ByteDance-Seed/Bagel/commit/cec368eedead3afc9f6c9c7bce3da0b86ce93ec9)
> 
>  for idx in range(start_idx + 1, end_idx + 1): 
>      instruction = random.choice(row["instruction_list"][idx-1]) 
>      data = self._add_text(data, instruction, need_loss=False) 
>      if idx != end_idx: 
>          data = self._add_image( 
>              data,  
>              pil_img2rgb(Image.open(io.BytesIO(row["image_list"][idx]))), 
>              need_loss=True,  
>              need_vae=True,  
>              need_vit=True, 
>          ) 
>      else: 
>          data = self._add_image( 
>              data,  
>              pil_img2rgb(Image.open(io.BytesIO(row["image_list"][idx]))), 
>              need_loss=True,  
>              need_vae=False,  
>              need_vit=False, 
>          )

get! thanks for ur reply!

**Comment by hychaochao** [comment]
I noticed that in the editing dataset, the text instruction data doesn’t contribute to the loss. However, during finetuning on the Generation & Editing with Thinking dataset, the text thought data should be included in the loss calculation — is that correct?

**Comment by hychaochao** [comment]
@eternaldolphin For example, the dataset with text thought mentioned in the paper. 

<img width="1332" alt="Image" src="https://github.com/user-attachments/assets/5d3e1fbb-7126-4acb-8c4a-529333db7dce" />

**Comment by Andy1621** [comment]
Yes, please check Section 3.3 in the paper.

**Comment by eternaldolphin** [comment]
except the thinking mode, do you have other format interleaved dataset, i mean that can output both text and image(instead of only input)
